cloudinit/config/schema.py

```diff
@@ -617,9 +617,10 @@ def validate_cloudconfig_file(config_path, schema, annotate=False):
                 "Unable to read system userdata as non-root user."
                 " Try using sudo"
             )
-        paths = read_cfg_paths()
-        user_data_file = paths.get_ipath_cur("userdata_raw")
-        content = load_file(user_data_file, decode=False)
+        init = Init(ds_deps=[])
+        init.fetch(existing="trust")
```

Comment:
As this branch is executed when the user requests to check the system config, and we can assume that `cloud-init` did run, then I think it would be safe to remove the following lines:

```python
        init.fetch(existing="trust")
        init.consume_data()
```

as the user-data has already been consumed.

Reply:
I think we should keep it as-is. When using the pickled data, the time to "fetch" is trivial when waiting on the command line.